implementations/rust/ockam/ockam_api/src/cloud/project.rs

```diff
@@ -65,43 +68,54 @@ mod tests {
 
     #[ockam_macros::test]
     async fn basic_api_usage(ctx: &mut Context) -> ockam_core::Result<()> {
+        // Create a Vault to safely store secret keys for Receiver.
+        let vault = Vault::create();
+
+        // Create an Identity to represent Receiver.
+        let receiver = Identity::create(&ctx, &vault).await?;
+
+        // Create an Identity to represent the ockam-command client.
+        let client_identity = Identity::create(&ctx, &vault).await?;
+
+        // Create an AuthenticatedStorage to store info about Receiver's known Identities.
+        let storage = InMemoryStorage::new();
+
+        // Create a secure channel listener for Receiver that will wait for requests to
+        // initiate an Authenticated Key Exchange.
+        receiver
+            .create_secure_channel_listener("api", TrustEveryonePolicy, &storage)
+            .await?;
```

Comment:
Since this block is used in both tests and production code, could we refactor it out as a function? I'd put it into `ockam_api/util.rs` for the time being.

Reply:
the receiver/listener is used only on tests on these commands,   and seems on nodes/services.rs but I _think_  the usage there would differ enough to not be worth trying to share it  (like,  it will load a real identity while for tests we just want to create a transient one, etc).   However,  between tests agree there is significant duplication, will try to  factor that out :+1:

Reply:
ok my bad, I thought I saw that logic in non-tests code more than once. Thanks!